### PR TITLE
Branch/launch txs

### PR DESCRIPTION
**Pull request by eric79**
This creates a `sim` route in the server for POST requests to launch a simulation on the server machine. Parameters in the request are the name of the application to launch, a command line argument to use when launching it, and a path to a json file that contains input data for the simulation.  It's not entirely general because it assumes that there will always be a command line argument and file path to be passed in.  But this will do for now until we need something more general.  :smile: 

@davideaster @landersk61 @youngca @AmbientOSX, would you mind reviewing?

There is a corresponding app branch named `launch-txs` that can be tested and code reviewed at the same time as this.

### CODE REVIEW

File: lib/nodejs/vwf.js [hunk 1]
```diff
@@ -139,19 +141,37 @@ function HandleFileRequest( request, response ) {
 // directory structure.
 // If all that fails, serve up a 404 response since the request was not handled.
 function Serve( request, response ) {
+    lobby.post( "/sim", function( req, res ) {
```
Comment (eric79) [inline lib/nodejs/vwf.js]: @davideaster I forgot that you said that I should move this into `lobby.js`.  I can do that in a future PR (unless I get to it before you review this).  :wink:

Comment (davideaster) [inline lib/nodejs/vwf.js]: Yes, I think that's a good idea. Be sure to consider where it goes relative to authentication. If it goes after, you would need to send authentication data in with the request (which is correct, but maybe ok to defer).

Comment (davideaster) [inline lib/nodejs/vwf.js]: For the other comments, the function would need `req, res, next` parameters.

File: lib/nodejs/vwf.js [hunk 2]
```diff
@@ -139,19 +141,37 @@ function HandleFileRequest( request, response ) {
...
+        var simDirectory = "sim";
+        var data = req.body;
+        var filePath = simDirectory + "/" + data.filePathFromSim;
+        fs.writeFile( filePath, data.simData, function( err ) {
+            if ( err ) {
+                return console.error( err );
```
Comment (davideaster) [inline lib/nodejs/vwf.js]: The handler should either complete the request using something like `res.sendStatus( ... )` or pass by calling `next`. The error can be forwarded to the containing application using `next( err )`. You can forward the error and exit this function by calling `return next( err )`. You can log the error here too, but it's probably fine to report it in the request, unless it relates to an unexpected error in the handler.

File: lib/nodejs/vwf.js [hunk 3]
```diff
@@ -139,19 +141,37 @@ function HandleFileRequest( request, response ) {
...
+            }
+            execFile( data.simExe, [
+                data.commandLineFlag,
+                data.filePathFromSim
+            ], {
+                cwd: simDirectory
+            }, function( err ) {
+                if ( err ) {
+                    return console.error( err );
```
Comment (davideaster) [inline lib/nodejs/vwf.js]: Ditto: `return next( err )`.

File: lib/nodejs/vwf.js [hunk 4]
```diff
@@ -139,19 +141,37 @@ function HandleFileRequest( request, response ) {
...
+                }
+            } );
+        } );
+        res.sendStatus( 200 );
```
Comment (davideaster) [inline lib/nodejs/vwf.js]: Sending the result here will mark every request successful regardless of whether errors eventually occur. The request also may or may not complete before the `writeFile` and the `execFile` complete. Calling `sendStatus` in the non-error case in the execFile callback will send successful completion after the app launches successfully.

**Comment by davideaster** [comment]
Looks pretty good. Once the callbacks are sorted out, I think it'll be ready.

**Comment by eric79** [comment]
@davideaster thanks for the comments.  What do you think now?

**Comment by davideaster** [comment]
Looks good. :+1:

**Pull request merged**